File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/GanttChartDataGrid/CustomDatesAndDragging/CustomGanttChartItem.cs
using DlhSoft.Windows.Controls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace Demos.WPF.CSharp.GanttChartDataGrid.CustomDatesAndDragging
{
    class CustomGanttChartItem : GanttChartItem, INotifyPropertyChanged
    {
        private DateTime customStart = DateTime.Today;
        public DateTime CustomStart
        {
            get { return customStart; }
            set
            {
                if (value > CustomFinish)
                    CustomFinish = value;
                customStart = value;
                OnPropertyChanged(nameof(CustomStart));
            }
        }
        public double ComputedCustomBarLeft
        {
            get { return GanttChartView.GetPosition(CustomStart) - GanttChartView.GetPosition(Start); }
        }

        private DateTime customFinish = DateTime.Today;
        public DateTime CustomFinish
        {
            get { return customFinish; }
            set
            {
                if (value < CustomStart)
                    CustomStart = value;
                customFinish = value;
                OnPropertyChanged(nameof(CustomFinish));
            }
        }
        public double ComputedCustomBarWidth
        {
            get { return GanttChartView.GetPosition(CustomFinish) - GanttChartView.GetPosition(CustomStart); }
        }

        protected override void OnPropertyChanged(string propertyName)
        {
            base.OnPropertyChanged(propertyName);
            switch (propertyName)
            {
                case nameof(Start):
                case nameof(CustomStart):
                    OnPropertyChanged(nameof(ComputedCustomBarLeft));
                    break;
            }
            switch (propertyName)
            {
                case nameof(CustomStart):
                case nameof(CustomFinish):
                    OnPropertyChanged(nameof(ComputedCustomBarWidth));
                    break;
            }
        }

        // Alternatively (required, if you have mouse wheel zooming enabled), refresh the user interface from a central handler.
        protected override void OnBarChanged()
        {
            OnPropertyChanged(nameof(ComputedCustomBarLeft));
            OnPropertyChanged(nameof(ComputedCustomBarWidth));
            base.OnBarChanged();
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples/WPF-VisualBasic/GanttChartDataGrid/CustomDatesAndDragging/CustomGanttChartItem.vb
Imports DlhSoft.Windows.Controls
Imports System.ComponentModel
Imports System.Text

Friend Class CustomGanttChartItem
    Inherits GanttChartItem
    Implements INotifyPropertyChanged

    Private customStartValue As Date = Date.Today
    Public Property CustomStart() As Date
        Get
            Return customStartValue
        End Get
        Set(value As Date)
            If value > CustomFinish Then
                CustomFinish = value
            End If
            customStartValue = value
            OnPropertyChanged(NameOf(CustomStart))
        End Set
    End Property
    Public ReadOnly Property ComputedCustomBarLeft() As Double
        Get
            Return GanttChartView.GetPosition(CustomStart) - GanttChartView.GetPosition(Start)
        End Get
    End Property

    Private customFinishValue As Date = Date.Today
    Public Property CustomFinish() As Date
        Get
            Return customFinishValue
        End Get
        Set(value As Date)
            If value < CustomStart Then
                CustomStart = value
            End If
            customFinishValue = value
            OnPropertyChanged(NameOf(CustomFinish))
        End Set
    End Property
    Public ReadOnly Property ComputedCustomBarWidth() As Double
        Get
            Return GanttChartView.GetPosition(CustomFinish) - GanttChartView.GetPosition(CustomStart)
        End Get
    End Property

    Protected Overrides Sub OnPropertyChanged(propertyName As String)
        MyBase.OnPropertyChanged(propertyName)
        Select Case propertyName
            Case NameOf(Start), NameOf(CustomStart)
                OnPropertyChanged(NameOf(ComputedCustomBarLeft))
        End Select
        Select Case propertyName
            Case NameOf(CustomStart), NameOf(CustomFinish)
                OnPropertyChanged(NameOf(ComputedCustomBarWidth))
        End Select
    End Sub

    ' Alternatively (required, if you have mouse wheel zooming enabled), refresh the user interface from a central handler.
    Protected Overrides Sub OnBarChanged()
        OnPropertyChanged(NameOf(ComputedCustomBarLeft))
        OnPropertyChanged(NameOf(ComputedCustomBarWidth))
        MyBase.OnBarChanged()
    End Sub
End Class
